### DISCUSSION TITLE
Is there a way for the MediaElement to deal with DRM?

### DISCUSSION THREAD

**Discussion by PeterVanVogelpoel** [CommunityToolkit/Maui#920] [Q&A]
Is there a way for the MediaElement to deal with DRM?

**Comment by jfversluis** [comment]
Not at the moment, if you have any ideas around it, feel free to open (or use this one) discussion with a feature request. I'd mostly be curious around the API you would have in mind for using this.

**Reply by PeterVanVogelpoel** [reply]
We are building a VOD platform and the mobile app should be able to play DRM. As far as I understand the MediaElement uses the ExoPlayer for Android. The ExoPlayer is capable of playing DRM. Shouldn't it be the case for the MediaElement as well then?
If we don't find a way to play DRM with the MAUI app that we build so far, we have to move to Flutter. We love MAUI in many ways. This is the one thing we really need that is missing. Maybe someone knows the way how to do it.

**Reply by jfversluis** [reply]
Sure, it is possible, we just haven’t made it available yet on the .NET MAUI layer. You have a couple of options:

* wait for us to implement it, as mentioned; please help us understand better how you would like to use this
* do a pull request and implement it yourself 
* Use the existing customization options in .NET MAUI to customize this control to your needs 

Lots of options! In regard to going to Flutter; please choose the right tool for the job, by all means. Although from what I understood in terms of media, it might not be: https://twitter.com/gaetschwartz/status/1617686528813322242

**Reply by PeterVanVogelpoel** [reply]
We are creating a multi device, cross platform, open kind of market place for independent filmmakers. For the web we use Blazor and we'd love to finish the mobile part with .NET MAUI. The mobile app should be able to deliver protected content (AES and/or DRM) and to do Chromecasting. In the backend Azure Media Services deals with the content.
You expect these functionalities out of the box in the near future?
In your video you mention sending custom headers for accessing protected resources. Could you point me to some documentation there?
Anyhow you are doing a great job, so we really hope we can do all we need with MAUI in the end.

**Reply by alwazirconnect** [reply]
any solution for play drm widevine in .net maui i need also to play my video content protected by windivine???

**Reply by jfversluis** [reply]
> You expect these functionalities out of the box in the near future?

I can't really say much about the timeline. This project is run in our own time so it's very much dependant on how much time I can find to work on this. If it's something that is really important to you, you could consider looking at implementing it yourself.

> In your video you mention sending custom headers for accessing protected resources. Could you point me to some documentation there?

I've started work on that in this branch: https://github.com/CommunityToolkit/Maui/tree/mediaelement-customheaders however, Android seems to not be able to change these headers without recreating the player, so that will be interesting... Not sure if we can send custom headers with Windows & Tizen at all.

> any solution for play drm widevine in .net maui i need also to play my video content protected by windivine???

No other solution than what was already mentioned at this time.

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889
